### PR TITLE
Automated pipeline for evaluating pre-computed fingerprints

### PR DESCRIPTION
**Pull request by blazejba**
Included scripts:
- `analyze_results.py` - it uses W&B API to access fingerprinting sweeps, and then, for each sweep it finds the "absolutely" best test score and the "fair" best test score (using the lowest validation loss as a proxy).
- `download_datasets.sh` - downloads all LargeMix datasets and splits and saves them under the path that the existing configs expect them to exist. 
- `finetune_on_fingerprints.py` - takes a file with fingerprints and a TDC dataset name and trains a small model, reporting the results to W&B. The script takes a bunch of arguments that allow customizing the model and training. This script is called multiple times when running sweeps. Everything is run on the CPUs and is built independently from Graphium.
- `sweep_checkpoint_on_all_tdc.py` - sweeps all 22 TDC datasets, uses `finetune_on_fingerprints.py` as the main program and the sweep is based on the parameters defined in `finetune_on_fingerprints_config.yaml`.
- `graphium/cli/get_final_fingerprints.py` - allows extraction of fingerprints by running all molecules from TDC through a selected checkpoint. Requires a checkpoint and an associated config. Everything is run on the CPUs. Requires a lot of RAM for larger models. 
- `graphium/nn/architectures/global_architectures.py` - includes changes that are necessary to run `graphium/cli/get_final_fingerprints.py`. 

The workflow:
1) [1 day - 2 weeks] Pretrain a model in Graphium 
2) [5 minutes - few hours] Extract fingerprints using `graphium/cli/get_final_fingerprints.py`; an explanation of how to use it can be found in this issue -> https://github.com/graphcore-research/scaling-molecular-gnns/issues/45
3) [5 days] Sweep on all TDC using `sweep_checkpoint_on_all_tdc.py`
4) [minutes] Analyze results using `analyze_results.py`
5) Update the results in the table (ask @blazejba for access)

### CODE REVIEW

**Review by kerstink-GC — COMMENTED** [review]
Looks good to me
